src/Microsoft.AspNetCore.Server.Kestrel.Core/Transport/IConnectionContext.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport
+{
+    public interface IConnectionContext
+    {
+        string ConnectionId { get; }
+        IPipeWriter Input { get; }
+        IPipeReader Output { get; }
+
+        // TODO: Remove these (Use Pipes instead?)
+        Task StopAsync();
+        void Abort(Exception ex);
+        void SetBadRequestState(RequestRejectionReason reason);
```

Comment:
I told you about this Thursday, and you told me to keep it lol. Obvs should change though.